### PR TITLE
Feature/idcom 1717  build with recent blockhash and sign

### PR DESCRIPTION
**Pull request by kevinhcolgan**
Change behaviour of build transaction to use a real blockhash and signature. signAndSendTransaction is changed to sendSerializedTransaction

### CODE REVIEW

File: solana/gatekeeper-lib/src/service/GatekeeperService.ts [hunk 1]
```diff
@@ -88,26 +87,26 @@ export class GatekeeperService {
...
    * @returns {Promise<string>}
    */
-  private async serializeBuiltTransaction(
+  private async signAndSerializeBuiltTransaction(
     transaction: Transaction
   ): Promise<string> {
     // recent blockhash and feepayer are required for serialization
-    // a real recent blockhash will be added in signAndSendTransaction
-    transaction.recentBlockhash = dummyBlockhash;
+    // a real recent blockhash will be added in sendSerializedTransaction
```
Comment (flipscholtz) [inline solana/gatekeeper-lib/src/service/GatekeeperService.ts:94]: Is this comment still accurate? It's adding the real blockhash now?

File: solana/gatekeeper-lib/src/service/GatekeeperService.ts [hunk 2]
```diff
@@ -88,26 +87,26 @@ export class GatekeeperService {
...
+    const recentBlockhash = await this.connection.getRecentBlockhash();
+    transaction.recentBlockhash = recentBlockhash.blockhash;
     transaction.feePayer = this.payer.publicKey;
-    const unsignedSerializedTx = await transaction
+    await transaction.sign(this.payer, this.gatekeeperAuthority);
+    const serializedTx = await transaction
       .serialize({
         verifySignatures: false,
```
Comment (flipscholtz) [inline solana/gatekeeper-lib/src/service/GatekeeperService.ts:101]: Should we now be setting these to true since we have the signature?

Comment (kevinhcolgan) [reply solana/gatekeeper-lib/src/service/GatekeeperService.ts]: you're right, updated

File: solana/gatekeeper-lib/src/service/GatekeeperService.ts (hunk 1 above)
Comment (kevinhcolgan) [reply solana/gatekeeper-lib/src/service/GatekeeperService.ts]: removed comment

**Review by flipscholtz — APPROVED** [review]

**Pull request merged**